File: restaurant_pos/lib/models/order.dart
import 'order_item.dart';

class OrderModel {
  final String id;
  final String restaurantId;
  final String? branchId;
  final String tableId;
  final String? customerId;
  final String? voucherId;
  final int? createdByUserId;
  final String invoiceNo;
  final String lookupCode;
  final String? type;
  final String? status;
  final String? paymentMethod;
  final String? paymentStatus;
  final String? paymentContent;
  final String? paymentReference;
  final int itemCount;
  final double subTotal;
  final double discountAmount;
  final double vatAmount;
  final double totalAmount;
  final String? notes;
  final DateTime? paidAtUtc;
  final DateTime createdAtUtc;

  final List<OrderItem> items;

  OrderModel({
    required this.id,
    required this.restaurantId,
    this.branchId,
    required this.tableId,
    this.customerId,
    this.voucherId,
    this.createdByUserId,
    required this.invoiceNo,
    required this.lookupCode,
    this.type,
    this.status,
    this.paymentMethod,
    this.paymentStatus,
    this.paymentContent,
    this.paymentReference,
    required this.itemCount,
    required this.subTotal,
    this.discountAmount = 0.0,
    this.vatAmount = 0.0,
    required this.totalAmount,
    this.notes,
    this.paidAtUtc,
    required this.createdAtUtc,
    required this.items,
  });

  factory OrderModel.fromJson(Map<String, dynamic> json) {
    return OrderModel(
      id: json['id']?.toString() ?? '',
      restaurantId: json['restaurantId']?.toString() ?? '',
      branchId: json['branchId']?.toString(),
      tableId: json['tableId']?.toString() ?? '',
      customerId: json['customerId']?.toString(),
      voucherId: json['voucherId']?.toString(),
      createdByUserId: (json['createdByUserId'] as num?)?.toInt(),
      invoiceNo: json['invoiceNo'] ?? '',
      lookupCode: json['lookupCode'] ?? '',
      type: json['type'],
      status: json['status'],
      paymentMethod: json['paymentMethod'],
      paymentStatus: json['paymentStatus'],
      paymentContent: json['paymentContent'],
      paymentReference: json['paymentReference'],
      itemCount: (json['itemCount'] as num?)?.toInt() ?? 0,
      subTotal: (json['subTotal'] as num?)?.toDouble() ?? 0.0,
      discountAmount: (json['discountAmount'] as num?)?.toDouble() ?? 0.0,
      vatAmount: (json['vatAmount'] as num?)?.toDouble() ?? 0.0,
      totalAmount: (json['totalAmount'] as num?)?.toDouble() ?? 0.0,
      notes: json['notes'],
      paidAtUtc: json['paidAtUtc'] != null ? DateTime.tryParse(json['paidAtUtc']) : null,
      createdAtUtc: DateTime.tryParse(json['createdAtUtc'] ?? '') ?? DateTime.now(),
      items: (json['items'] as List?)?.map((i) => OrderItem.fromJson(i)).toList() ?? [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'restaurantId': restaurantId,
      'branchId': branchId,
      'tableId': tableId,
      'customerId': customerId,
      'voucherId': voucherId,
      'createdByUserId': createdByUserId,
      'invoiceNo': invoiceNo,
      'lookupCode': lookupCode,
      'type': type,
      'status': status,
      'paymentMethod': paymentMethod,
      'paymentStatus': paymentStatus,
      'paymentContent': paymentContent,
      'paymentReference': paymentReference,
      'itemCount': itemCount,
      'subTotal': subTotal,
      'discountAmount': discountAmount,
      'vatAmount': vatAmount,
      'totalAmount': totalAmount,
      'notes': notes,
      'paidAtUtc': paidAtUtc?.toIso8601String(),
      'createdAtUtc': createdAtUtc.toIso8601String(),
      'items': items.map((i) => i.toJson()).toList(),
    };
  }

  // Compatibility getter
  DateTime get dateTimeUtc => createdAtUtc;
}

File: restaurant_pos_web/backend/Models/Order.cs
using System.ComponentModel.DataAnnotations;

namespace RestaurantPos.Api.Models;

public class Order
{
    [Key]
    public string Id { get; set; } = Guid.NewGuid().ToString("N");
    public string RestaurantId { get; set; } = string.Empty;
    public string? BranchId { get; set; }
    public string TableId { get; set; } = string.Empty;
    public string? CustomerId { get; set; }
    public string? VoucherId { get; set; }
    public int? CreatedByUserId { get; set; }
    public string InvoiceNo { get; set; } = string.Empty;
    public string LookupCode { get; set; } = string.Empty;
    public string? Type { get; set; }
    public string? Status { get; set; }
    public string? PaymentMethod { get; set; }
    public string? PaymentStatus { get; set; }
    public string? PaymentContent { get; set; }
    public string? PaymentReference { get; set; }
    public int ItemCount { get; set; }
    public decimal SubTotal { get; set; }
    public decimal DiscountAmount { get; set; }
    public decimal VatAmount { get; set; }
    public decimal TotalAmount { get; set; }
    public string? Notes { get; set; }
    public DateTime? PaidAtUtc { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;

    public List<OrderItem> Items { get; set; } = [];
}
